alerts/cluster-network-operator/V4SubnetAllocationThresholdExceeded.md

```diff
@@ -0,0 +1,40 @@
+# V4SubnetAllocationThresholdExceeded
+
+## Meaning
+
+The `V4SubnetAllocationThresholdExceeded` alert is triggered when more than
+80% of subnets are being allocated for the hostsubnet.
+
+## Impact
+
+This is a warning alerts. The subnets will be exhausted and no more worker nodes
+can be added to the cluster if it keeps growing. And eventually, no more
+workloads can be provisioned.
+
+## Diagnosis
+
+Check the network configuration on the cluster.
+
+    oc get networks.config.openshift.io/cluster -o jsonpath='{.spec.clusterNetwork}'
+
+    [{"cidr":"10.128.0.0/14","hostPrefix":23}]
+
+Calculate the IPv4 subnets capability.
+
+    subnet_capability = 2^[(32 - clusternetwork_netmask) - (32 - hostPrefix)]
```

Comment:
we dont need to do this - we have metrics to show used and total subnet capacity:
ovnkube_master_allocated_v4_host_subnets
ovnkube_master_num_v4_host_subnets 
I think its easier to subtract those two numbers to get remaining node capacity.

total nodes that can be added to the cluster = ovnkube_master_num_v4_host_subnets - ovnkube_master_allocated_v4_host_subnets

Reply:
We are using the metrics to trigger the alerts. Counting from the cluster provides an alternative way to confirm the fact, IMO.

And it is easier to count from cluster and config directly instead of login to the prometheus to get the metrics?

Reply:
yeah, fair enough. Thanks Bo.